### PR TITLE
Add support for ES API keys

### PR DESCRIPTION
**Pull request by larskuhtz**

### CODE REVIEW

**Review by mightybyte — COMMENTED** [review]
Looks reasonable to me but we should get another review or two.

File: src/Utils/Logging/Config.hs [hunk 1]
```diff
@@ -130,19 +134,27 @@ handleConfigFromText x = case CI.mk x of
     "stdout" -> return StdOut
     "stderr" -> return StdErr
     _ | CI.mk (T.take 5 x) == "file:" -> return $ FileHandle (T.unpack (T.drop 5 x))
-    _ | CI.mk (T.take 3 x) == "es:" -> return $ ElasticSearch (T.drop 3 x)
+    _ | CI.mk (T.take 4 x) == "es::" -> return $ ElasticSearch (T.drop 4 x) Nothing
+    _ | CI.mk (T.take 7 x) == "es:http" -> return $ ElasticSearch (T.drop 3 x) Nothing
+        -- legathy format without API key (deprecated)
```
Comment (slpopejoy) [inline src/Utils/Logging/Config.hs:139]:
```suggestion
        -- legacy format without API key (deprecated)
```

Comment (slpopejoy) [reply src/Utils/Logging/Config.hs:139]: Also it's unclear which case this refers to. Assuming it's the above, the comment should probably go above

File: src/Utils/Logging.hs [hunk 2]
```diff
@@ -587,15 +623,127 @@ withElasticsearchBackend
...
+
+    _putLog req a = req
+        { HTTP.method = "POST"
+        , HTTP.path = HTTP.path req <> T.encodeUtf8 streamName <> "/_doc"
+        , HTTP.responseTimeout = HTTP.responseTimeoutMicro 3000000
+        , HTTP.requestHeaders = ("content-type", "application/json") : HTTP.requestHeaders req
+        , HTTP.requestBody = HTTP.RequestBodyLBS $ encode $ JsonLogMessage a
+        }
+
+    putBulgLog req a = req
```
[the remark this answers is not in the document]
Comment (larskuhtz) [reply src/Utils/Logging.hs:705]: yes

File: src/Utils/Logging.hs [hunk 3]
```diff
@@ -587,15 +623,127 @@ withElasticsearchBackend
...
+          where
+            isTimeout = Nothing <$ (readTVar timer >>= check)
+            fill = tryReadTBQueue queue >>= maybe retry (return . Just)
+
+        go 0 !batch _ = return (0, batch)
+        go !remaining !batch !timer = getNextAction timer >>= \case
+            Nothing -> return (remaining, batch)
+            Just x -> go (remaining - 1) (batch <> indexAction x) timer
+
+    _createDataStream req =
```
[the remark this answers is not in the document]
Comment (larskuhtz) [reply src/Utils/Logging.hs:688]: Yes, I added a comment for why we keep it around.

File: src/Utils/Logging.hs (hunk 2 above)
[the remark this answers is not in the document]
Comment (larskuhtz) [reply src/Utils/Logging.hs:703]: done

**Review by slpopejoy — APPROVED** [review]

**Pull request merged**